### PR TITLE
APEXCORE-471 Reissue of the resource was failing for the BlackListBas…

### PR DESCRIPTION
**Pull request by sandeshh**
…ed scheduler, it happened because RequestedResource was never empty. Issue is fixed after making the operation similar to that of ResourceRequestHandler with the handling for Blacklist

### CODE REVIEW

File: engine/src/main/java/com/datatorrent/stram/BlacklistBasedResourceRequestHandler.java [hunk 1]
```diff
@@ -98,7 +100,19 @@ public void reissueContainerRequests(AMRMClient<ContainerRequest> amRmClient, Ma
     }
   }
 
-  public void recreateContainerRequest(Map<StreamingContainerAgent.ContainerStartRequest, MutablePair<Integer, ContainerRequest>> requestedResources, int loopCounter, ResourceRequestHandler resourceRequestor, List<ContainerRequest> removedContainerRequests)
+  @Override
+  public void addContainerRequest(Map<StreamingContainerAgent.ContainerStartRequest, MutablePair<Integer, ContainerRequest>> requestedResources, int loopCounter, List<ContainerRequest> containerRequests, StreamingContainerAgent.ContainerStartRequest csr, ContainerRequest cr)
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/BlacklistBasedResourceRequestHandler.java]: Why is the change necessary?

Comment (sandeshh) [reply engine/src/main/java/com/datatorrent/stram/BlacklistBasedResourceRequestHandler.java]: I have done 2 changes apart from fixing the bug

1. Reduced the access scope of function calls 
2. Grouped the overridden method together

It improves the code readability, but if you think it is subjective then I can revert 1 or 2 or both.

Comment (vrozov) [reply engine/src/main/java/com/datatorrent/stram/BlacklistBasedResourceRequestHandler.java]: +1 to reduce the access scope.
-0 to move methods unless their implementation is modified.

Comment (sandeshh) [reply engine/src/main/java/com/datatorrent/stram/BlacklistBasedResourceRequestHandler.java]: Done.

**Review by vrozov — APPROVED** [review]
Changes look good to me. Will merge in couple days.

**Pull request merged**